open-sphere-base/core/src/main/java/io/opensphere/core/NetworkConfigurationManager.java

```diff
@@ -4,19 +4,24 @@
 public interface NetworkConfigurationManager
 {
     /**
-     * The "purpose" that should be used when creating a username/password
-     * provider for the network proxy.
+     * The "purpose" that should be used when creating a username/password provider for the network proxy.
      */
     String PROXY_PURPOSE = "proxy";
 
     /**
-     * Add a listener for changes to the network configuration. Only a weak
-     * reference is held.
+     * Add a listener for changes to the network configuration. Only a weak reference is held.
      *
      * @param listener The listener.
      */
     void addChangeListener(NetworkConfigurationChangeListener listener);
 
+    /**
+     * Add a listener for changes to the proxy settings. Only a weak reference is held.
+     *
+     * @param listener the listener
+     */
+    void addChangeListener(ProxySettingsChangeListener listener);

```

Comment:
The ProxySettingsChangeListener does the same thing as the NetworkConfigurationChangeListener I suggest just using the NetworkConfigurationChangeListener and remove ProxySettingsChangeListener.

Reply:
I tried using NetworkConfigurationChangeListener and it fires way too many times - maybe once for each change? The new listener fires once to notify that the apply button has been hit and only fires once.

Reply:
Could I maybe get rid of all the calls to notifyNetworkConfigChanged() in NetworkConfigurationManagerImpl and call the method from NetworkConfigurationOptionsProvider in applyChanges() once? The only other thing listening for changes is HttpServerProvider and I don't think it needs to be notified multiple times to clear servers.